app/models/mixins/cloud_tenancy_mixin.rb

```diff
@@ -13,7 +18,8 @@ def tenant_id_clause_format(tenant_ids)
     end
 
     def tenant_joins_clause(scope)
-      scope.includes(:cloud_tenant => "source_tenant").includes(:ext_management_system)
+      scope.includes(QUERY_REFERENCES)
+           .references(QUERY_REFERENCES) # needed for the where to work
```

Comment:
do we need to also make this change over to the other `tenant_joins_clause` code? e.g.:
`app/models/flavor.rb#tenant_joins_clause` ?

Reply:
Damn, you might be right, and it looks like we don't have a test for that either... which is how I caught this in the first place... 😑

Reply:
> # needed for the where to work

As someone who doesn't look at query optimizations all day, I'm not sure what this means.  Why is this needed to make the `where` work?  Which `where`?  Can we better explain why?

Reply:
If read correctly, maybe saying something about being explicit instead of relying on `references(nil)` to blindly do the right thing sometimes...

Reply:
@jrafanie yeah, looking at this diff on it's own makes it hard to see, but it is for the other method in this short file, `#tenant_id_clause_format`, to function properly.

So maybe I should change it to:

```ruby
# # needed for the where with `tenant_id_clause_format` to work
```

?

Reply:
Yeah, but say what "to work" / function properly means... N+1 in spite of `includes`?  Does it fail to build the sql?  How is the existing code working or not working here?

Reply:
Means the SQL is invalid without it.

The method in this file, `tenant_id_clause_format`, has columns that "reference" the other tables it is including

```ruby
def tenant_id_clause_format(tenant_ids)
"(tenants.id IN (?) AND ext_management_systems.tenant_mapping_enabled IS TRUE)..."
end
```

And I explain how is was not failing briefly in my [second commit](https://github.com/ManageIQ/manageiq/pull/17141/commits/ab09c7ca3) (which adds the `.skip_references` bit to begin with).  Specifically, this is the explanation from that commit:

> The change to the CloudTenancyMixin was made so that it can be in charge
> of it's own specific references instead of relying on a quirk of
> `ActiveRecord`, in which `references(nil)` will call references for all
> of the `includes` values.  Also a few models that overwrote the
> tenant_joins_clause method had to be updated as well.